src/FileIO.cpp

```diff
@@ -580,5 +580,24 @@ namespace FileIO {
 
       return (0 == rc);
    }
+
+   Result<std::vector<std::string>> GetDirectoryContents(const std::string& directory) {
+     std::vector<std::string> filesInDirectory;
+     DIR* dir;
+     struct dirent* entry;
+     if ((dir = opendir (directory.c_str())) != NULL) {
+       while ((entry = readdir (dir)) != NULL) {
+         if (!(std::string(entry->d_name) == "." || std::string(entry->d_name) == "..")) {
```

Comment:
I think you would be better off using the `d_type == DT_REG`. That way you would only include regular files in the vector. And you might call this function, GetDirectoryRegularFiles.